src/System.Private.CoreLib/src/System/Runtime/CompilerServices/ConditionalWeakTable.cs

```diff
@@ -424,6 +433,22 @@ internal bool Remove(TKey key)
             }
 
             //----------------------------------------------------------------------------------------
+            // Removes all entries from the table.
+            //
+            // Precondition:
+            //      Must hold _lock.
+            //----------------------------------------------------------------------------------------
+            internal void Clear()
+            {
+                // Remove all handles (as in Remove, just by setting the hashcode and leaving
+                // actual removal up to the finalizer).
+                for (int i = 0; i < _firstFreeEntry; i++)
```

Comment:
Actually, I can undo this clear change, here and in coreclr. If changed it when the finalizer was killing out the parent's container, as then this dropped container could me with its replacement, but I changed the finalizer to only null it out when it matches this, so it should be fine again.